### PR TITLE
docs: add public Git error messages

### PR DESCRIPTION
**Pull request by angeloashmore**
## Context

DT-1941

## The Solution

Add Markdown files for each Git-specific error. We will link to the public GitHub pages to allow users to learn more about an error.

We can use the `prismic.dev/msg` domain to alias links:

- **Source**: https://prismic.dev/msg/slicemachine/main/git-error
- **Destination**: https://github.com/prismicio/slice-machine/blob/main/messages/git-error.md

In the future, we may move these error messages to a prismic.io-hosted page.

## Impact / Dependencies

The GitHub integration must link to the correct page. See https://github.com/prismicio/sm-api/pull/7.

## Checklist before requesting a review

- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: messages/git-error.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Git: Error
+
+An error occured while syncing models to Prismic from a Git repository.
```
Comment (bapmrl) [inline messages/git-error.md]: ⚠️ Typo: "occurred".

File: messages/git-error.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+### How to fix the issue
+
+Syncing the project again by committing to the repository.
```
Comment (bapmrl) [inline messages/git-error.md]: 💡 Maybe "Sync" would be better?

File: messages/git-invalid-slice-machine-config.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Git: Invalid Slice Machine config
+
+The Git repository's Slice Machine configuration file (`slicemachine.config.json`) was not valid.
+
+Any of the following issues could cause this error:
+
+- Missing required properties, such as `repositoryName` and `adapter`.
+- Invalid values given to configuration properties.
+- Invalid JSON with incorrect syntax
```
Comment (bapmrl) [inline messages/git-invalid-slice-machine-config.md]: ⚠️ Missing "." at the end.

File: messages/git-unauthorized-github.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Git: Unauthorized Github access
```
Comment (bapmrl) [inline messages/git-unauthorized-github.md]: ⚠️ Typo: "GitHub".

File: messages/git-unauthorized-prismic.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Git: Unauthorzed Prismic access
```
Comment (bapmrl) [inline messages/git-unauthorized-prismic.md]: ⚠️ Typo: "Unauthorized".

File: messages/git-unauthorized-prismic.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+The GitHub app tried to make a request to Prismic, but failed due to unauthorized access.
+
+This error can occur when syncing model changes to Prismic from GitHub with an invalid Prismic Write API token.
+
+## How to fix the issue
+
+First, ensure a Prismic Write API token was saved in Slice Machine. You can set, update, and delete a Write API token from Slice Machine's Settings page.
```
Comment (bapmrl) [inline messages/git-unauthorized-prismic.md]: ⚠️ This sentence isn't 100% exact as the token is only asked in the `GitRepositoryConnectDialog` when connecting to a repository. The user needs to disconnect the repository in order to be able to update the token (when connecting again).

File: messages/git-unsupported-adapter.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# Git: Unsupported adapter
+
+The Slice Machine project uses an adapter that is not supported in the GitHub app.
+
+The following adapters are supported:
+
+- [`@slicemachine/adapter-next`](https://github.com/prismicio/slice-machine/tree/main/packages/adapter-next#readme): Next.js projects
```
Comment (bapmrl) [inline messages/git-unsupported-adapter.md]: ❓ In a previous file you used `.` at the end of list items. What style should be adopted (for the sake of consistency)?

**Review by bapmrl — CHANGES_REQUESTED** [review]
Excellent!

File: messages/git-unauthorized-prismic.md (hunk 6 above)
Comment (angeloashmore) [reply messages/git-unauthorized-prismic.md]: Thanks, I updated it to these instructions:

```md
## How to fix the issue

Re-connect the Prismic and GitHub repositories, ensuring the Write API token is active and correct:

1. In Slice Machine's Settings page, disconnect the connected GitHub repository.
1. Re-connect the GitHub repository.
1. When asked for a Write API token, provide a valid token.

Tokens can be created following these steps:

1. In the Prismic repository, navigate to the Settings page.
1. Select **API & Security** in the sidebar and the **Write APIs** tab at the top of the page.
1. Create and copy a new Write API token or copy an existing token.
```

File: messages/git-unsupported-adapter.md (hunk 7 above)
Comment (angeloashmore) [reply messages/git-unsupported-adapter.md]: Thanks, I added periods to each item.

In general, I follow this pattern:

- If list items are sentences, use periods.
- If list items only contain specific subjects, don't use periods.

With periods:

> Follow these guidelines:
> - Use apples liberally.
> - Avoid bananas when possible.
> - Puree carrots and use for something else.

Without periods:

> The following fruits may be used:
> - Apple
> - Banana
> - Carrot

**Review by bapmrl — APPROVED** [review]

**Pull request merged**